CMakeLists.txt

```diff
@@ -0,0 +1,137 @@
+cmake_minimum_required(VERSION 3.26...3.30)
+
+# Define the variables necessary to build and install the project differently depending on whether
+# the project is being configured from scikit-build-core. If configured from outside
+# scikit-build-core, we need to use placeholder values for the variables that are usually defined
+# by scikit-build-core.
+if(DEFINED SKBUILD_PROJECT_NAME)
+    set(CLP_FFI_PY_INSTALL_LIBS ON CACHE BOOL "Enable installing the built libraries." FORCE)
+    set(CLP_FFI_PY_PROJECT_NAME
+        ${SKBUILD_PROJECT_NAME}
+        CACHE STRING
+        "The name of the project parsed by scikit-build-core."
+        FORCE
+    )
+    set(CLP_FFI_PY_VERSION
+        ${SKBUILD_PROJECT_VERSION}
+        CACHE STRING
+        "The version of the project parsed by scikit-build-core."
+        FORCE
+    )
+else()
+    set(CLP_FFI_PY_INSTALL_LIBS OFF CACHE BOOL "Disable installing the built libraries." FORCE)
+    set(CLP_FFI_PY_PROJECT_NAME "clp-ffi-py" CACHE STRING "Use a placeholder project name." FORCE)
+    set(CLP_FFI_PY_VERSION "0.0.0" CACHE STRING "Use a placeholder version." FORCE)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS
+        ON
+        CACHE BOOL
+        "Enable/Disable output of compile commands during generation."
+        FORCE
+    )
+endif()
+
+project(${CLP_FFI_PY_PROJECT_NAME} LANGUAGES CXX VERSION ${CLP_FFI_PY_VERSION})
+
+# C-extension modules will be compiled as shared libraries. To enable dynamic linking, all libraries
+# (including the static ones) must be compiled with the position-independent-code option.
+set(CMAKE_POSITION_INDEPENDENT_CODE ON CACHE BOOL "Compile libraries using PIC.")
+
+find_package(
+    Python
+    REQUIRED
+    COMPONENTS
+        Interpreter
+        Development.Module
+)
+
+set(CLP_FFI_PY_LIB_IR "native")
+python_add_library(
+    ${CLP_FFI_PY_LIB_IR}
+    MODULE
+    WITH_SOABI
+)
+
+target_compile_features(${CLP_FFI_PY_LIB_IR} PRIVATE cxx_std_20)
+
+set(CLP_FFI_PY_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
+set(CLP_FFI_PY_LIB_SRC_DIR "${CLP_FFI_PY_SRC_DIR}/clp_ffi_py")
+set(CLP_CORE_ROOT "${CLP_FFI_PY_SRC_DIR}/clp/components/core")
```

Comment:
As discussed offline and in the comment from the previous review, this should probably be `CLP_FFI_PY_CLP_CORE_SRC_DIR`.

Reply:
Renamed to `CLP_FFI_PY_CLP_CORE_DIR`. Strictly speaking the source directory is under core directory, so it's not `SRC_DIR`